crates/db_connection_pool/src/dbconnection/snowflakeconn.rs

```diff
@@ -191,6 +195,34 @@ pub fn snowflake_schema_cast(record_batch: &RecordBatch) -> Result<RecordBatch,
                 columns.push(cast_sf_timestamp_ntz_to_arrow_timestamp(column)?);
                 continue;
             }
+
+            let is_decimal_type = matches!(
+                field.data_type(),
+                DataType::Decimal128(_, _) | DataType::Decimal256(_, _)
+            );
+
+            if !is_decimal_type && sf_logical_type.to_lowercase().as_str() == "fixed" {
+                if let (Some(precision_str), Some(scale_str)) =
+                    (field_metadata.get("precision"), field_metadata.get("scale"))
+                {
+                    if let (Ok(precision), Ok(scale)) =
+                        (precision_str.parse::<u8>(), scale_str.parse::<i8>())
```

Comment:
**question:** What other arrow `DataType`s do we expect to receive here?

Reply:
Possible DataTypes here are Int8, 16, 32, etc.